### ISSUE TITLE
Faced error - Value Error: Unknown Error.Try using MMD-based detection instead

### ISSUE THREAD

**Issue by Rithika1000** [SeldonIO/alibi-detect#801]
[E_Commerce_Churn_Kaggle.zip](https://github.com/SeldonIO/alibi-detect/files/11596476/E_Commerce_Churn_Kaggle.zip)
This file has the data that i have used. I read it as a csv using pandas ( E_Commerce_Dataset-Baseline.csv as ref_df and E_Commerce_Dataset-Production.csv as test_df) and then implemented the below code.

First i preprocessed the data as follows
`
categorical_feature_name=['Preferred_Device', 'Payment_Mode',
       'Gender', 'Prefered_Category']

def Encoder(df):
  le = LabelEncoder()
  for i in categorical_feature_name:
      encoded_df = le.fit_transform(df[[i]])
      df.drop([i], axis=1, inplace=True)
      df[i]=encoded_df
 return df
ref_df = Encoder(ref_df)
test_df = Encoder(test_df)

x_ref = np.asarray(ref_df, np.float32)
x_test = np.asarray(test_df, np.float32)
`

I then calculated drift using the following code:
`
lsdd1 = LSDDDrift(x_ref, p_val=0.05, backend='pytorch')

lsdd1_pred = lsdd1.predict(x_test)
stat = 'LSDD value='
stat_val, p_val = lsdd1_pred['data']['distance'], lsdd1_pred['data']['p_val']
print(f'{stat} {stat_val:.3f} -- p-value {p_val:.3f}')
`
i then got the following error
![image](https://github.com/SeldonIO/alibi-detect/assets/123930151/59c19dbd-74ff-479b-8f78-47da01ff34f8)
![image](https://github.com/SeldonIO/alibi-detect/assets/123930151/a72d34cc-6a5a-470c-a40f-0518fda6aa56)

**Comment by jklaise** [comment]
Hi, can you please format the code using Github code blocks? It's very hard to read otherwise. https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Comment by jklaise** [comment]
@Rithika1000 I've chatted to my team and have been advised that the LSDD method is only appropriate for detecting drift on continuous distributions. Looking [at the docs](https://docs.seldon.io/projects/alibi-detect/en/latest/cd/methods/lsdddrift.html#Overview) this makes sense as densities only exist for continuous data, but your example features categorical data which the method would not work with.

We will add some more information to our docs so that it's more clear for people in the future about this limitation, thanks for reporting!

**Comment by Rithika1000** [comment]
Thanks!!

**Comment by jklaise** [comment]
@Rithika1000 I'll refer you to this comment: https://github.com/SeldonIO/alibi-detect/pull/802#issuecomment-1568640968 Long story short - MMD can be used with one-hot encoded categorical variables and the method will work, but generally for tabular data specifically we recommend using the `TabularDrift` detector.
